topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -137,46 +154,205 @@ We don't want to overload our training VMs trying to run real tools, so to demon
+>         assign: ['db-skip-locked']
+>       execution:
+>    -    default: singularity
+>    +    default: tpv_dispatcher
+>         environments:
+>    -      local_env:
+>    -        runner: local_runner
+>    -        tmp_dir: true
+>    -      slurm:
+>    -        runner: slurm
+>    -        singularity_enabled: true
+>    -        env:
+>    -        - name: LC_ALL
+>    -          value: C
+>    -        - name: SINGULARITY_CACHEDIR
+>    -          value: /tmp/singularity
+>    -        - name: SINGULARITY_TMPDIR
+>    -          value: /tmp
+>    -      singularity:
+>    -        runner: local_runner
+>    -        singularity_enabled: true
+>    -        env:
+>    -        # Ensuring a consistent collation environment is good for reproducibility.
+>    -        - name: LC_ALL
+>    -          value: C
+>    -        # The cache directory holds the docker containers that get converted
+>    -        - name: SINGULARITY_CACHEDIR
+>    -          value: /tmp/singularity
+>    -        # Apptainer uses a temporary directory to build the squashfs filesystem
+>    -        - name: SINGULARITY_TMPDIR
+>    -          value: /tmp
+>    +      tpv_dispatcher:
+>    +        runner: dynamic
+>    +        type: python
+>    +        function: map_tool_to_destination
+>    +        rules_module: tpv.rules
+>    +        tpv_config_files:
+>    +          - "{{ tpv_config_dir }}/tpv_rules_local.yml"
+>       tools:
+>         - class: local # these special tools that aren't parameterized for remote execution - expression tools, upload, etc
+>           environment: local_env
+>    @@ -138,6 +119,8 @@ galaxy_config_files_public:
+>     galaxy_config_files:
+>       - src: files/galaxy/themes.yml
+>         dest: "{{ galaxy_config.galaxy.themes_config_file }}"
+>    +  - src: files/galaxy/config/tpv_rules_local.yml
+>    +    dest: "{{ tpv_mutable_dir }}/tpv_rules_local.yml"
+>     
+>     galaxy_config_templates:
+>       - src: templates/galaxy/config/container_resolvers_conf.yml.j2
+>    {% endraw %}
+>    ```
+>    {: data-commit="Add TPV to job config"}
+>
+>  Note that we set the default execution environment to the tpv_dispatcher, added the tpv_dispatcher itself as a dynamic runner, and removed all other destinations.
+>  Adding TPV as a runner will cause Galaxy to automatically install the `total-perspective-vortex` package on startup as a conditional dependency.
+>  Finally, we added a new config file named `tpv_rules_local.yml`, which we will create next.
 >
-> 1. Edit your `templates/galaxy/config/job_conf.yml.j2` and add the following destination. Then, map the new tool to the new destination using the tool ID (`<tool id="testing">`) and destination id (`<destination id="slurm-2c">`) by adding a new section to the job config, `<tools>`, below the destinations:
+> 2. Create a new file named `tpv_rules_local.yml` in the "files" folder of your ansible playbook, so that it is copied to the config folder on the target.
```

Comment:
```suggestion
> 2. Create a new file named `tpv_rules_local.yml` in the `files/galaxy/config/` folder of your ansible playbook, so that it is copied to the config folder on the target.
```